api/src/main/java/org/apache/cloudstack/api/command/admin/storage/SeedOfficialSystemVMTemplateCmd.java

```diff
@@ -0,0 +1,166 @@
+        responseObject = ZoneResponse.class, requestHasSensitiveInfo = false, responseHasSensitiveInfo = false, since = "4.13",
+        authorized = {RoleType.Admin})
+public class SeedOfficialSystemVMTemplateCmd extends BaseCmd {
+
+    public Long getId() {
+        return id;
+    }
+
+    @Parameter(name = ApiConstants.ID, type = CommandType.UUID, entityType = ZoneResponse.class, required = true, description = "the ID of the zone")
+    private Long id;
+
+    public String getUrl() {
+        return url;
+    }
+
+    @Parameter(name = ApiConstants.URL, type = CommandType.STRING, entityType = ZoneResponse.class, description = "The template url")
+    private String url;
+
+    @Parameter(name = ApiConstants.HYPERVISOR, type = CommandType.STRING, entityType = ZoneResponse.class, description = "The Hypervisor to register this template for")
+    private String hypervisor;
+
+    public Long getTemplateId() {
+        return templateId;
+    }
+
+    public void setTemplateId(Long templateId) {
+        this.templateId = templateId;
+    }
+
+    private Long templateId;
+
+    public String getInstallPath() {
+        return installPath;
+    }
+
+    public void setInstallPath(String installPath) {
+        this.installPath = installPath;
+    }
+
+    private String installPath;
+
+    public String getFileExtension() {
+        return fileExtension;
+    }
+
+    public void setFileExtension(String fileExtension) {
+        this.fileExtension = fileExtension;
+    }
+
+    private String fileExtension;
+
+
+    @Override
+    public void execute(){
+        // Get all the clusters for this zone
+        Pair<List<? extends Cluster>, Integer> clusters = _mgr.searchForClusters(this);
+        HashSet<String> hypervisors = new HashSet<>();
+        // we need to get the hypervisors from the clusters
+        for (Cluster cluster: clusters.first()){
+            hypervisors.add(cluster.getHypervisorType().toString());
+        }
+
+        // we need to get secondary storage for this zone.
+        HashSet<String> imageStores = _queryService.searchForImageStores(this);
+
+        String script = Script.findScript("scripts/storage/secondary/","cloud-install-sys-tmplt");
+
+        // mount locally
+        String mountPoint = "/tmp/nfsmount";
+        Script.runSimpleBashScript("mkdir " + mountPoint);
+
+        for (String imageStore: imageStores){
+            try {
+                URI uri = new URI(imageStore);
+                Script.runSimpleBashScript("sudo mount -t nfs " + uri.getHost() + ":" + uri.getPath() + " " + mountPoint);
+            } catch (URISyntaxException e) {
+                throw new CloudRuntimeException("Malformed URI " + imageStore);
+            }
+
+            // now download the template to the image store for every hypervisor if not specified
+            if (this.hypervisor != null && this.url != null){
+
+                if (this.hypervisor.equalsIgnoreCase("kvm") || this.hypervisor.equalsIgnoreCase("lxc")) {
```

Comment:
Check `Hypervisor.HypervisorType.getType(hypervisor)` function. You can convert the String to a Hypervisor type and compare using that type